lib/maze/checks/check.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Maze
+  # Base class for data verification checks.  This is designed to be an abstraction on top of Minitest::Assertions,
+  # allowing checks to be implemented in different ways (e.g. no op and delayed fail).
+  class Check
```

Comment:
Yeah, I'm not sure what the best option is for this kind of thing in Ruby TBH

It made me in nervous in this case because defaulting to noop an assertion could cause tests to pass when they shouldn't, if a method wasn't implemented by `AssertionCheck` but was in `NoOpCheck`

Reply:
https://refactoring.guru/design-patterns/factory-method/ruby/example#example-0 uses the approach of throwing `NotImplementedError` from the base class methods, which could be used to guard against that.  For this, though, I'm not sure there's actually much benefit in having a base class.